test/extended/openstack/utils.go

```diff
@@ -108,6 +141,147 @@ func DeleteNamespace(ctx context.Context, clientSet *kubernetes.Clientset, ns *v
+		return nil
+	}
+	e2e.Logf("PersistentVolumeClaim %v was created", pvc.ObjectMeta.Name)
+
+	return pvc
+}
+
+func waitPvcVolume(ctx context.Context, clientSet *kubernetes.Clientset, pvc string, ns string) error {
+	err := wait.PollUntilContextTimeout(ctx, 15*time.Second, 3*time.Minute, true, func(ctx context.Context) (bool, error) {
+		pvc, err := clientSet.CoreV1().PersistentVolumeClaims(ns).Get(ctx, pvc, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		} else {
+			if pvc.Spec.VolumeName != "" {
+				return true, nil
+			} else {
+				return false, nil
+			}
+		}
+	})
+	return err
+}
+
+// return share from openstack with specific name
+func GetSharesFromName(client *gophercloud.ServiceClient, shareName string) ([]shares.Share, error) {
+	var emptyShare []shares.Share
+
+	listOpts := shares.ListOpts{
+		Name: shareName,
+	}
+	allPages, err := shares.ListDetail(client, listOpts).AllPages()
+	if err != nil {
+		return emptyShare, err
+	}
+	shares, err := shares.ExtractShares(allPages)
+	if err != nil {
+		return emptyShare, err
+	}
+	return shares, nil
+}
+
+func FindStorageClassByProvider(oc *exutil.CLI, provisioner string) *storagev1.StorageClass {
+	scList, err := oc.AdminKubeClient().StorageV1().StorageClasses().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			e2e.Logf("no storage classes found")
+			return nil
+		}
+		e2e.Failf("could not list storage classes: %v", err)
+	}
+	for _, sc := range scList.Items {
+		//if strings.HasPrefix(sc.Provisioner, provisionerPrefix) {
+		//		e2e.Logf("ignoring in-tree storage class %s", sc.Name)
+		//		continue
+		//}
+		if sc.Provisioner == provisioner {
+			return &sc
+		}
+	}
+	e2e.Logf("no default storage class found")
+	return nil
+}
+
+// Creates *appsv1.Deployment using the image agnhost configuring a server on the specified port and protocol
+// Further info on: https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+func createTestDeployment(deploymentOpts *DeploymentOpts) *appsv1.Deployment {
+
+	isFalse := false
+	isTrue := true
+
+	var netExecParams string
+	var volumes []v1.Volume
+	var volumeMounts []v1.VolumeMount
+
+	switch deploymentOpts.Protocol {
+	case v1.ProtocolTCP:
+		netExecParams = fmt.Sprintf("--http-port=%d", deploymentOpts.Port)
+	default:
+		netExecParams = fmt.Sprintf("--%s-port=%d", strings.ToLower(string(deploymentOpts.Protocol)), deploymentOpts.Port)
+	}
+
+	testDeployment := e2edeployment.NewDeployment(deploymentOpts.Name, deploymentOpts.Replicas, deploymentOpts.Labels, "test",
+		imageutils.GetE2EImage(imageutils.Agnhost), appsv1.RollingUpdateDeploymentStrategyType)
+	testDeployment.Spec.Template.Spec.SecurityContext = &v1.PodSecurityContext{}
+	testDeployment.Spec.Template.Spec.Containers[0].SecurityContext = &v1.SecurityContext{
+		Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+		AllowPrivilegeEscalation: &isFalse,
```

Comment:
Have you tried `ptr.To(false)` instead? (import `"k8s.io/utils/ptr"` first)

Reply:
Done